electrum/network.py

```diff
@@ -625,18 +628,29 @@ def _set_default_server(self) -> None:
         assert isinstance(self.default_server, ServerAddr), f"invalid type for default_server: {self.default_server!r}"
 
     def _set_proxy(self, proxy: Optional[dict]):
-        self.proxy = proxy
-        dns_hacks.configure_dns_resolver()
-        self.logger.info(f'setting proxy {proxy}')
+        if self.proxy == proxy:
+            return
 
+        self.logger.info(f'setting proxy {proxy}')
+        self.proxy = proxy
         self.tor_proxy = False
```

Comment:
Shouldn't this be tri-state instead, and start with None? That way you can potentially distinguish False and dont-know.
(also btw, I see this is a pre-existing name, but I think `is_proxy_tor` or `is_tor_proxy` would be a better name)

Reply:
agreed, done.